maven-model-builder/src/main/java/org/apache/maven/model/interpolation/StringSearchModelInterpolator.java

```diff
@@ -45,10 +44,14 @@
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
+import javax.inject.Named;
+import javax.inject.Singleton;
+
 /**
  * StringSearchModelInterpolator
  */
-@Component( role = ModelInterpolator.class )
+@Named
+@Singleton
```

Comment:
Will this properly determine the role?

Reply:
Yes; it's also the [advice in Sisu/PlexusMigration](https://wiki.eclipse.org/Sisu/PlexusMigration).

Reply:
How, if I look at the Type Override section with `extends` it explicitly references the interface name.

Reply:
From that document:
> though in some rare cases an explicit type is required.
>...
> `extends SomeSupportClassHardToGuessTypeFrom`

My impression from the code (is it [this part](https://github.com/eclipse/sisu.inject/blob/f4d69f725f23cc3b7dabedc5ee8d0fbec665add8/org.eclipse.sisu.inject/src/org/eclipse/sisu/space/QualifiedTypeBinder.java#L407)?) is that a simple type hierarchy with interfaces is fine, even if it goes via another class. That note is about more complex cases than this.

As a test, if I add:

    @Typed(AbstractStringBasedModelInterpolator.class)

to intentionally mis-bind, the test fails (with no `ModelInterpolator`). So I'm confident it's being properly determined here.